src/telemetry/metrics/mod.rs

```diff
@@ -106,17 +106,17 @@ struct Metric<'a, M: FmtMetric> {
 }
 
 /// The root scope for all runtime metrics.
-#[derive(Debug, Default)]
+#[derive(Clone, Debug, Default)]
 struct Root {
-    requests: http::RequestScopes,
-    responses: http::ResponseScopes,
-    transports: transport::OpenScopes,
-    transport_closes: transport::CloseScopes,
+    requests: Arc<Mutex<http::RequestScopes>>,
```

Comment:
I'm still reviewing, but is there a use case for having an `Arc` per field, instead of `Arc<Root>`?

Reply:
probably not in this state. I'm trying to drive the Arcs down to the leaves so that the sensor actually holds the Arc<Scope> and the registry holds a Weak reference to it (so the registry doesnt need to attempt to manage metric lifetimes itself)